spec/support/factory_girl.rb

```diff
@@ -20,4 +20,12 @@
     end_year "1999"
     genre "cartoon"
   end
+
+  factory :review do
+    rating 5
+    sequence(:body) { |n| "Celtics are #{n} champions (but really 17 times)" }
+    score 0
+    association :program
```

Comment:
You should just be able to put `program` and `user` by themselves on 28 & 29, and factory girl will create the associated dependent objects.